### PR TITLE
Avoid an unnecesary git clone for core repositories if clone_repositories is called later anyhow

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: buildkite/build_project.sh [hunk 1]
```diff
@@ -57,8 +57,18 @@ case "$REPO_FULL_NAME" in
...
             latest_tag="IS-ALREADY-CHECKED-OUT"
         else
-            # otherwise checkout the respective branch
-            latest_tag=$("$DIR/origin_target_branch.sh" "${REPO_URL}")
+            # for the main core repositories,
+            # clone_repositories.sh will clone them together
+            case "$REPO_FULL_NAME" in
+                "dlang/dmd" | \
+                "dlang/druntime" | \
+                "dlang/phobos")
```
Comment (jacob-carlborg) [inline buildkite/build_project.sh]: tools and dub as well?

Comment (wilzbach) [reply buildkite/build_project.sh]: No they are checked out as individual repos as they don't depend on the core repos - just an up to date DMD which is always built initially and already provided ty them.

Comment (wilzbach) [reply buildkite/build_project.sh]: This is just about these three repos as for them clone_repositories will be executed during "project testing"

Comment (jacob-carlborg) [reply buildkite/build_project.sh]: Ok.

**Pull request merged**